stacktrace.go

```diff
@@ -53,33 +51,33 @@ type StacktraceFrame struct {
 
 // Try to get stacktrace from err as an interface of github.com/pkg/errors, or else NewStacktrace()
 func GetOrNewStacktrace(err error, skip int, context int, appPackagePrefixes []string) *Stacktrace {
-	stacktracer, errHasStacktrace := err.(interface {
-		StackTrace() errors.StackTrace
-	})
-	if errHasStacktrace {
-		var frames []*StacktraceFrame
-		for f := range stacktracer.StackTrace() {
-			pc := uintptr(f) - 1
-			fn := runtime.FuncForPC(pc)
-			var fName string
-			var file string
-			var line int
-			if fn != nil {
-				file, line = fn.FileLine(pc)
-				fName = fn.Name()
-			} else {
-				file = "unknown"
-				fName = "unknown"
-			}
-			frame := NewStacktraceFrame(pc, fName, file, line, context, appPackagePrefixes)
-			if frame != nil {
-				frames = append([]*StacktraceFrame{frame}, frames...)
-			}
-		}
-		return &Stacktrace{Frames: frames}
-	} else {
+	type stackTracer interface {
+		StackTrace() []runtime.Frame
```

Comment:
In respect of `/pkg/errors` this implementation does not work. Not yet sure if I'm doing something wrong, but passing error created by `errors.New(...)` to this function, ends up in `!ok` if statement. This is because `pkg/errors` says:
```
// New, Errorf, Wrap, and Wrapf record a stack trace at the point they are
// invoked. This information can be retrieved with the following interface:
//
//     type stackTracer interface {
//             StackTrace() errors.StackTrace
//     }
//
// The returned errors.StackTrace type is defined as
//
//     type StackTrace []Frame
```
How should I use this method with `/pkg/errors` ? Im confused.